webviz_config/_theme_class.py

```diff
@@ -75,6 +75,33 @@ def adjust_csp(self, dictionary: dict, append: bool = True) -> None:
             else:
                 self._csp[key] = value
 
+    def update_plotly_layout(self, layout: dict) -> dict:
```

Comment:
I guess the name is inspired by current usage, e.g. `dict_a.update(dict_b)`, where `dict_a` is updated with the content of `dict_b`. Here we instead (I guess) want to merge two dictionaries (the plotly layout in the theme, and the plugin/user provided plotly layout), and return the new object? Should we use e.g. `merge`/`combine`/`...` instead of `update`?

Reply:
I would say that we are still updating `dict_a` (the theme layout) with the contents of `dict_b` (the layout for the specific case), where contents of `dict_b` is prioritized over `dict_a` if it exists in both (so that you can force deviations from the theme). `merge` and `combine` sounds more like they are two equally prioritized dicts during the process.

Reply:
In addition we create a new dict/object I guess (i.e. not only mutate `dict_a` as in `dict_a.update`)?

Reply:
Aah, ok, now I see your concern. Sounds a bit like you update the `plotly theme layout`. What about something like `create_themed_layout`? Then it should be clear that we are making a completely new `dict`?

Reply:
Good name suggestion 👍